### PR TITLE
Bump `oracle`

### LINKED ISSUE #660: Bump `oracle`
To:

- `forc 0.42.1`
- `fuel-core 0.18.2`
- `fuels-rs 0.43.0`

### PR DESCRIPTION
**Pull request by K1-R1**
## Type of change

- Improvement (refactoring, restructuring repository, cleaning tech debt, ...)
- Documentation

## Changes

The following changes have been made:

- Bump application to `forc 0.42.1` and `fuel-core 0.18.2`
- Bump test harness to `fuels 0.43.0`

## Related Issues

Closes #660

### CODE REVIEW

File: oracle/README.md [hunk 1]
```diff
@@ -59,56 +59,56 @@ The project can be started by executing the following steps:
...
+| WALLET_SECRET      | Private key of the first deterministic wallet provided by the [fuels-rs](https://github.com/FuelLabs/fuels-rs) sdk. This private key correspondes to the `owner` address specified in the oracle contract's [`Forc.toml`](./project/contracts/oracle-contract/Forc.toml). This address is also configured in step 4 to have the maximum amount of the [BASE_ASSET](https://github.com/FuelLabs/sway/blob/master/sway-lib-std/src/constants.sw).                                                                                             |
+| FUEL_PROVIDER_URL  | Fuel-core network url normally set as http://localhost:4000/graphql for development.                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
```
Comment (Braqzen) [inline oracle/README.md:111]: In terms of the preview it looks exactly the same.
Now we have a bunch of new characters to visually format the table in non-preview mode.

From 1 perspective this is useless, from another it helps a little.
I'm somewhat ambivalent regarding this change so your call.

File: oracle/project/fuel-toolchain.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [toolchain]
-channel = "latest-2023-04-12"
+channel = "latest-2023-07-07"
 
 [components]
-forc = "0.37.0"
-fuel-core = "0.17.9"
+forc = "0.42.1"
+fuel-core = "0.18.2"
```
Comment (Braqzen) [inline oracle/project/fuel-toolchain.toml:6]: According to [this release](https://github.com/FuelLabs/fuelup/blob/gh-pages/channels/latest/channel-fuel-latest-2023-07-07.toml#L81) we are 1 patch ahead. I'm unsure of the reason behind your choice of 18.2 but I'll leave it to you to bump or leave it at this

**Review by Braqzen — APPROVED** [review]

File: oracle/README.md (hunk 1 above)
Comment (K1-R1) [reply oracle/README.md:111]: I didn't do this manually, not sure what formatter made the change. I don't think it needs to be changed.

File: oracle/project/fuel-toolchain.toml (hunk 2 above)
Comment (K1-R1) [reply oracle/project/fuel-toolchain.toml:6]: I chose the same version of `fuel-core` that `forc 0.42.1` uses

**Pull request merged**